queue/priority_queue.go

```diff
@@ -120,8 +120,9 @@ func (pq *PriorityQueue) Put(items ...Item) error {
 	}
 
 	pq.lock.Lock()
+	defer pq.lock.Unlock()
```

Comment:
In our experience, defer has a performance overhead. Have you re-run the benchmarks on `Put` to see how they are affected?